src/app_service/service/contacts/service.nim

```diff
@@ -327,20 +352,22 @@ QtObject:
 
     if(pubkey == singletonInstance.userProfile.getPubKey()):
       # If we try to get the contact details of ourselves, just return our own info
-      return ContactsDto(
-        id: singletonInstance.userProfile.getPubKey(),
-        displayName: singletonInstance.userProfile.getDisplayName(),
-        name: singletonInstance.userProfile.getPreferredName(),
-        alias: singletonInstance.userProfile.getUsername(),
-        ensVerified: singletonInstance.userProfile.getPreferredName().len > 0,
-        added: true,
-        image: Images(
-          thumbnail: singletonInstance.userProfile.getThumbnailImage(),
-          large: singletonInstance.userProfile.getLargeImage()
-        ),
-        trustStatus: TrustStatus.Trusted,
-        bio: self.settingsService.getBio(),
-        socialLinks: self.settingsService.getSocialLinks()
+      return self.constructContactDetails(
```

Comment:
Make me wonder, can't we save our own ContactDetails in a cache? Maybe here or in the singleton. That way we don't have to reprocess it each time.

Reply:
Good idea. Addressed: https://github.com/status-im/status-desktop/pull/10805/commits/6b690686657c6907d93d47d3ea6b0ea0ecc0eb44

Reply:
We need to make sure to update the cache when we update our profile though. It can be done in another PR if you want. (just remove the commit and put it in another branch)

Reply:
Right, haven't thought of it :+1: